src/lib.rs

```diff
@@ -465,24 +477,30 @@ fn pan_orbit_camera(
                 has_moved = true;
             }
         } else if scroll.abs() > 0.0 {
-            let apply_scroll = |value: f32| value - scroll * value * 0.2;
+            let delta = - scroll * pan_orbit.target_radius * 0.2;
+            pan_orbit.target_radius = util::apply_zoom_limits(
+                pan_orbit.target_radius + delta,
+                pan_orbit.zoom_upper_limit,
+                pan_orbit.zoom_lower_limit,
+            );
 
-            if let Projection::Orthographic(ref mut p) = *projection {
-                p.scale = util::apply_zoom_limits(
-                    apply_scroll(p.scale),
-                    pan_orbit.zoom_upper_limit,
-                    pan_orbit.zoom_lower_limit,
-                );
-            } else {
-                pan_orbit.radius = pan_orbit.radius.map(|radius| {
-                    util::apply_zoom_limits(
-                        apply_scroll(radius),
-                        pan_orbit.zoom_upper_limit,
-                        pan_orbit.zoom_lower_limit,
-                    )
-                });
+        }
+        if let Projection::Orthographic(ref mut p) = *projection {
```

Comment:
I can see you've moved this part out of the above block, so that it executes when there aren't scroll events (to keep approaching target). This is fine, but I think we should have it follow the same pattern as the others. In other words, put this `if let {} else {}` in section 4, and add a condition that checks if `pan_orbit.target_radius != radius`. You might want to put `has_moved = true` in the above `else if scroll.abs() > 0.0` block, but it might not be necessary (has_moved may not be necessary at all come to think of it but that's another topic).

I think following the same pattern makes it easier to understand overall.

Reply:
I've rearranged the code to make it more readable, and now `has_moved = true;` will be executed when `pan_orbit.target_radius != radius.`